### PR TITLE
Stream client interceptor

### PR DESCRIPTION
**Pull request by hothero**

### CODE REVIEW

File: interceptors/interceptors.go [hunk 1]
```diff
@@ -78,194 +80,7 @@ func DefaultClientInterceptor(address string) grpc.UnaryClientInterceptor {
 	return grpc_middleware.ChainUnaryClient(DefaultClientInterceptors(address)...)
 }
 
-//DebugLoggingInterceptor is the interceptor that logs all request/response from a handler
```
Comment (hothero) [inline interceptors/interceptors.go:81]: move to separate files: unary_interceptors.go and stream_interceptors.go

File: interceptors/interceptors.go [hunk 2]
```diff
@@ -78,194 +80,7 @@ func DefaultClientInterceptor(address string) grpc.UnaryClientInterceptor {
...
-		}
-		err = handler(srv, stream)
-		if !modifiers.HasDontLogError(ctx) {
-			notifier.Notify(err, ctx)
-		}
-		return err
-
-	}
+//DefaultStreamClientInterceptor are the set of default interceptors that should be applied to all client calls
+func DefaultStreamClientInterceptor() grpc.StreamClientInterceptor {
```
Comment (hothero) [inline interceptors/interceptors.go:84]: new function

File: interceptors/stream_interceptors.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+		if !modifiers.HasDontLogError(ctx) {
+			notifier.Notify(err, ctx)
+		}
+		return err
+
+	}
+}
+
+// ForwardMetadataInterceptor forwards metadata from upstream to downstream
+func ForwardMetadataStreamClientInterceptor() grpc.StreamClientInterceptor {
```
Comment (hothero) [inline interceptors/stream_interceptors.go:50]: new function

**Comment by ankurs** [comment]
Yay, finally some use of GRPC Streaming 🥳

**Pull request merged**